torch/distributed/tensor/debug/_visualize_sharding.py

```diff
@@ -1,177 +1,251 @@
 # mypy: allow-untyped-defs
+import importlib.util
 from collections.abc import Sequence
 
 import numpy as np
-
 from torch._prims_common import ShapeType
 from torch.distributed.tensor import DeviceMesh
 from torch.distributed.tensor.placement_types import Placement, Shard
 
-
 __all__ = ["visualize_sharding"]
 
+Color = tuple[float, float, float] | str
 
-def _mesh_to_coordinate(mesh, device_type):
-    """
-    Given a n-dimensional list of device mesh, this function creates a map of
-    device and its coordinate
-    """
-    # Convert the n-dimensional list to a NumPy array
-    np_mesh = np.array(mesh.mesh.tolist())
-
-    # Create a dictionary to map each value to its coordinate
-    device_to_coordinate_map = {}
-    for coord, value in np.ndenumerate(np_mesh):
-        # device is unique in device_mesh
-        device_to_coordinate_map[f"{device_type}:{str(value)}"] = list(coord)
-
-    return device_to_coordinate_map
 
+def _device_coords_in_mesh(mesh: DeviceMesh) -> dict[int, tuple[int, ...]]:
+    """Given a device mesh, returns a dict from device index to coordinate."""
+    return {
+        device_index: coord for coord, device_index in np.ndenumerate(np.array(mesh.mesh.tolist()))
+    }
 
-def _convert_offset_to_ranges(all_offsets):
-    """
-    Using tabulate package to create a table is easier when we specify row and col ranges
-    This function converts offsets to ranges.
-    """
-    converted_blocks = []
 
-    for offset in all_offsets:
-        shape, offset, value = offset
-
-        # Calculate row_range and column_range
-        row_range = (offset[0], offset[0] + shape[0] - 1)
-        column_range = (offset[1], offset[1] + shape[1] - 1)
-
-        # Convert value to string to match your desired format
-        converted_block = {
-            "row_range": row_range,
-            "column_range": column_range,
-            "value": str(value),
-        }
-        converted_blocks.append(converted_block)
-
-    return converted_blocks
-
-
-def _create_table(blocks):
+def _shard_info(
```

Comment:
I was glad to see `_compute_local_shape_and_global_offset` is deleted, but it seems like it is mostly copy-pasted into the new `_shard_info` fn.  If possible, i'd rather you find a way to call the real `torch.distributed._tensor._utils.compute_local_shape_and_global_offset` function, which I recently updated to handle uneven-sharding case.  This way the logic wouldn't diverge.  (I wasn't even aware of this copy ... )